test/tests/services/filters-parser.js

```diff
@@ -0,0 +1,170 @@
+import Interface from 'forest-express';
+import FiltersParser from '../../../src/services/filters-parser';
+import mongooseConnect from '../../utils/mongoose-connect';
+import { InvalidFiltersFormatError, NoMatchingOperatorError } from '../../../src/services/errors';
+
+describe('Service > FiltersParser', () => {
+  let IslandModel;
+  let defaultParser;
+  const timezone = 'Europe/Paris';
+
+  const options = {
+    mongoose,
+    connections: [mongoose],
+  };
+
+  before(() => {
+    Interface.Schemas = {
+      schemas: {
+        Island: {
+          name: 'Island',
+          idField: 'id',
+          primaryKeys: ['id'],
+          isCompositePrimary: false,
+          searchFields: ['name'],
+          fields: [
+            { field: 'id', type: 'Number' },
+            { field: 'name', type: 'String' },
+            { field: 'size', type: 'Number' },
+            { field: 'isBig', type: 'Boolean' },
+            { field: 'inhabitedOn', type: 'Date' },
+          ],
+        },
+      },
+    };
+
+    return mongooseConnect()
+      .then(() => {
+        const IslandSchema = mongoose.Schema({
+          id: { type: Number },
+          name: { type: String },
+          size: { type: Number },
+          isBig: { type: Boolean },
+          inhabitedOn: { type: Date },
+        });
+
+        IslandModel = mongoose.model('Island', IslandSchema);
+
+        defaultParser = new FiltersParser(IslandModel, timezone, options);
+        return IslandModel.remove({});
+      })
+      .then(() => {
+        return loadFixture(IslandModel, [
+          {
+            // id: 100,
+            name: 'Pyk',
+            size: 13,
+            isBig: false,
+            inhabitedOn: '2019-02-11T23:00:00.000Z',
+          },
+          {
+            // id: 101,
+            name: 'Dragonstone',
+            size: 124,
+            isBig: true,
+            inhabitedOn: '2018-03-21T23:00:00.000Z',
+          },
+        ]);
+      });
+  });
+
+  after((done) => {
+    mongoose.connection.close();
+    done();
+  });
+
+  describe('formatAggregation function', () => {
+    it('should format correctly aggregated conditions', () => {
+      expect(defaultParser.formatAggregation('and', [
+        { name: new RegExp('.*y.*') },
+        { size: { $lt: 455 } },
+      ])).to.deep.equal({ $and: [{ name: new RegExp('.*y.*') }, { size: { $lt: 455 } }] });
+    });
+
+    it('should format correctly nested conditions', () => {
+      const formattedConditions = [
+        { size: { $gt: 56 } },
+        { $or: [{ name: new RegExp('^P.*') }, { isBig: true }] },
+      ];
+      expect(defaultParser.formatAggregation('and', formattedConditions)).to.deep.equal({ $and: formattedConditions });
+    });
+
+    it('should throw an error on empty condition', () => {
+      expect(() => defaultParser.formatAggregation()).to.throw(NoMatchingOperatorError);
+      expect(() => defaultParser.formatAggregation({})).to.throw(NoMatchingOperatorError);
```

Comment:
Test with missing keys

Reply:
Operator: 'low'

Reply:
This is not resolved. You need a test with a wrong operator

Reply:
`expect(() => defaultParser.formatAggregation({ aggregator: 'unknown' })).to.throw(NoMatchingOperatorError);`